docs/idm-concept.md

```diff
@@ -0,0 +1,206 @@
+    3. If Device Public Key is flagged as revoked, compare its revocation effective date with the signature date
+
+#### Managing application sessions
+
+Users will be able to revoke one or all application sessions from within the identity's applications list. After revoking a session, the application will no longer be able to use that session, effectively stopping it from being able to perform operations that require the Session and Device Private Keys.
+
+#### Revoking a device
+
+Users must be able to revoke a device associated with an identity. To do so, the DID-Document needs to be updated in order to signal that the Device Public Key was revoked, including the effective date which is important when verifying signatures.
+
+Depending on the DID-method and the permissions of the Device Private Key, higher level keys or other processes might be necessary to update the DID-Document, such as using the Master Private Key.
+
+The way a public key is declared as revoked is [yet to be standardized](https://github.com/w3c-ccg/did-spec/issues/96) in the DID spec. Nevertheless, a DID-Document may be augmented via the `@context` attribute to support having that information.
+
+#### Replication and synchronization
+
+The same identity might be held inside different IDM Wallets. For that reason, the identities' data, such as its credentials and applications, will be seamlessly replicated to other IDM Wallets using P2P technologies powered by [CRDTs](https://en.wikipedia.org/wiki/Conflict-free_replicated_data_type).
+
+Any data stored off-chain or on-chain as per the DID-method best-practices will be taken into consideration, meaning they will be synchronized accordingly.
+
+### IDM Wallet UI
+
+End users need a graphical user-interface to interact with their IDM Wallet. An IDM Wallet UI is an application that will make use of the most suitable IDM Wallet SDK for the programming language they choose.
+
+Even though anyone can implement its own IDM Wallet UI application, there is a need to define some standards in regards to user journeys, user experience and interactions. For that reason, there will be a reference IDM Wallet UI based on web-technologies to be used inside a browser, that can be packed as an [Electron](https://electronjs.org/) in the future.
+
+#### Reference UI diagrams
+
+User-journeys diagram:
+
+<img src="https://raw.githubusercontent.com/ipfs-shipyard/pm-idm/master/docs/images/diagram_user-journeys.png" alt="" width="1000">
+
+Information Architecture diagram:
+
+<img src="https://raw.githubusercontent.com/ipfs-shipyard/pm-idm/master/docs/images/diagram_information-architecture.png" alt="" width="1000">
+
+#### Reference UI concept
+
+The goal of the reference UI is to have a really polished interface with a premium feel. It must be easy to use, distinct, reliable, accessible and provide a feel of credibility to the users.

```

Comment:
Thank you for this write-up! Very helpful!

A few questions re: getting you useful feedback: Who is your target user group? Please add any information around who you're designing for, and any baseline research you did to understand their needs (or if this is a prototype without discovery research, then it would be great to have a problem statement around what you're optimizing for in this iteration, or what you're testing/prototyping). Once I know who the target is, or what you're aiming for with the prototype, I'll be able to provide better feedback.

Reply:
Understood, I will be talking to @andreforsousa and get back to you regarding this.

Reply:
@meiqimichelle We will be creating a SWOT and Audience analysis and make them part of this repo: https://github.com/ipfs-shipyard/pm-idm/issues/38

You may follow that issue to get informed about the progress. I will also update this concept once we have done that.

Thanks for the heads up.